Move FoldableCard component out from calypso submodule

harriswong opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->

## Description
Part of effort from https://github.com/Automattic/woocommerce-services/issues/2333, this PR moves `FoldableCard` out from the calypso's submodule. This PR includes the following changes:
- Added a new `client/components/foldable-card, a copy from the calypso submodule
- Added a new `client/components/screen-reader-text, a copy from the calypso submodule

### Related issue(s)
https://github.com/Automattic/woocommerce-services/issues/2333
<!--
  Is there an issue open this PR addresses? If so, link to it for more information.
  GitHub link example: "Fixes #1234" Syntax: `[close|closes|closed|fix|fixes|fixed|resolve|resolves|resolved] #1234`
  Note: Remove this section if this PR does not have a related issue.
-->

### Steps to reproduce & screenshots/GIFs
1. Make sure "Foldable Card" works. There are two types, compact, and not compact. 
2. To test, go to an order and create an label, the "step" should open/close properly: 
![image](https://user-images.githubusercontent.com/572862/124038166-efca7000-d9bd-11eb-8ab5-f1a9aecba80d.png)
3. Go to WooCommerce Services settings (`wp-admin/admin.php?page=wc-settings&tab=shipping&section=woocommerce-services-settings`), click "Add package" under "Packaging", these card should open/close properly:
![image](https://user-images.githubusercontent.com/572862/124038527-7bdc9780-d9be-11eb-9e87-fdd93f74cc02.png)



<!--
  Please include steps to reproduce, as well as screenshots or GIFs, showing the before & after.
  This helps expedite review and increase confidence for approval & merge.
-->

### Checklist

<!-- All testable code should have tests. -->
- [ ] unit tests

<!-- An entry in the changelog file is always required -->
- [ ] `changelog.txt` entry added

Commit:
Move FoldableCard component out from calypso submodule

harriswong commented on client/components/foldable-card/index.js:
```diff
@@ -0,0 +1,161 @@
+ /**
+  * Internal Dependencies
+  */
+ import { Card } from '@wordpress/components';
```
The current submodule's code uses `CompactCard` and `Card`, I changed this to use WordPress' components instead. 

The current submodule: https://github.com/Automattic/wp-calypso/blob/df4ddfdc94438b6d31b686d6415e08a567570944/client/components/foldable-card/index.jsx#L16-L17

brezocordero replied:
👍

harriswong commented on client/components/foldable-card/index.js:
```diff
@@ -0,0 +1,161 @@
+                     className="foldable-card__action foldable-card__expand"
+                     onClick={ clickAction }
+                 >
+                     <ScreenReaderText>{ screenReaderText }</ScreenReaderText>
```
I debated whether or not to copy over Calypso's `<ScreenReaderText`> component. I ended up copying it because:
- it has less change compared to the existing submodule
- other Calypso components may also need it
- we will clean up this entire `client/components` folder so then it's not introducing long term tech debt.

brezocordero replied:
`<ScreenReaderText>` has a different name on the WordPress components, it is called [VisuallyHidden](<ScreenReaderText>). 
This is [the issue where the component was added to Gutenberg and why the name was changed for alignment with other libraries](https://github.com/WordPress/gutenberg/pull/18022).

I am ok with copying it over, it is small and [the latest version on Calypso](https://github.com/Automattic/wp-calypso/tree/trunk/packages/components/src/screen-reader-text) is the same as our copy.

We can also create a separate issue to replace `<ScreenReaderText>`

harriswong replied:
If there is a WordPress component for this, maybe we should switch over to it. If the goal is to remove Calypso components completely, then I think it is beneficial to have as little calypso component dependency as possible.

harriswong replied:
Update: Removed ScreenReaderText component and use VisuallyHidden component instead.

brezocordero commented on client/components/foldable-card/index.js:
```diff
@@ -0,0 +1,161 @@
+     }
+
+     render() {
+         const itemSiteClasses = classNames( 'foldable-card', this.props.className, {
```
```suggestion
         const itemSiteClasses = classNames( 'foldable-card', 'card', this.props.className, {
```
The styles are a bit different when using the WordPress card. 
![NewFoldable](https://user-images.githubusercontent.com/8002881/124171168-d41ea280-da6d-11eb-8e9b-b09f80710994.png)
![CurrentFoldable](https://user-images.githubusercontent.com/8002881/124171187-d84ac000-da6d-11eb-9937-f962ad085dbc.png)


[On this PR](https://github.com/Automattic/woocommerce-services/pull/2337), we copied over the Calypso CSS for Card and then pass it to the WordPress Card. A bit redundant but it doesn't change the display.

harriswong replied:
Thanks! Updated.

brezocordero commented on client/components/foldable-card/style.scss:
```diff
@@ -0,0 +1,183 @@
+	position: relative;
+	transition: margin 0.15s linear;
+	padding: 0;
+
```
```suggestion
border-radius: 0;
```
By default, the WordPress card is rounded and it looks wavy when we have multiple cards. 

![RoundedCorners](https://user-images.githubusercontent.com/8002881/124175776-a3416c00-da73-11eb-8767-bf3b067e9cb3.png)

The newest version allows you to disable it with [isRounded={ false }](https://github.com/WordPress/gutenberg/blob/fe77cd3c7f9df1a5ac0a273097f7eb05282e638e/packages/components/src/card/card/component.js#L33) but that is not available yet.
Please take a second look in case adding the CSS line changes something else.

harriswong replied:
Added the `radius:0 change`.

brezocordero commented on client/extensions/woocommerce/app/order/order-activity-log/day.js:
```diff
@@ -9,7 +9,7 @@ import React, { Component } from 'react';
  * Internal dependencies
  */
-import FoldableCard from 'components/foldable-card';
+import FoldableCard from 'wcs-client/components/foldable-card';
```
Where can I test this?

harriswong replied:
This was introduced in https://github.com/Automattic/woocommerce-services/pull/2370#discussion_r591760370. If we want to test whether or not webpack found the right file (`client/components/foldable-card/index.js`), we could add a random class name in one of the divs and see if it renders.

brezocordero replied:
Thanks! Sorry my question was not clear. Where in the UI do I find the order-activity-log/day.js to test the changes?

harriswong replied:
I believe it is the notes on the order page itself: 
![image](https://user-images.githubusercontent.com/572862/124195872-7b103800-da88-11eb-9d85-fee2095943d0.png)

brezocordero reviewed (changes requested):
@harriswong works pretty well!
A couple of small comments to make the UI similar to what we have now.

Commit:
Update card border radius css

Commit:
Replace ScreenReaderText with VisuallyHidden component

harriswong commented:
@brezocordero I updated the PR with the following changes, please take another look again.
- Removed round corners for the cards
- Removed `ScreenReaderText` 
- Use `VisuallyHidden` component instead.

brezocordero reviewed (approved):
Thanks for making all the changes!
LGTM!
:shipit:
